File: truth-machine.scala
object TruthMachine {
    def loop(truth:Int) {
        println(truth)
        if (truth == 1) {
            loop(truth)
        }
    }

    def main(args: Array[String]) {
        loop(scala.io.StdIn.readLine().toInt)
    }
}

File: truth-machine.rs
use std::io;

fn main() {
    let mut truth = String::new();
    io::stdin().read_line(&mut truth).unwrap();
    let truth: i32 = truth.trim().parse().unwrap();
    loop {
        println!("{}", truth);
        if truth == 0 {
            break;
        }
    }
}
